### PR TITLE
7788 no else return False Negative fix

### LINKED ISSUE #7788: False negative for `no-else-return` after `except`
### Bug description

This snippet returns in the `except` block, and therefore the `else` is unnecessary:

```python
def f() -> bool:
    try:
        print('asdf')
    except:  # pylint: disable = bare-except
        return False
    else:
        return True
```

It should raise `no-else-return`.

### Configuration

_No response_

### Command used

```shell
`pylint asdf.py`
```

### Pylint output

```shell
No warnings raised
```

### Expected behavior

Should raise `no-else-return`

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
```

### PR DESCRIPTION
**Pull request by clavedeluna**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

`no else return` and `no else raise` now account for try/except/else pattern.

Closes #7788

### CODE REVIEW

**Comment by nickdrozd** [comment]
Tested locally; works!

**Comment by clavedeluna** [comment]
Primer output looks spot on! 🚀

**Comment by mbyrnepr2** [comment]
It may be good to add a test which uses `finally` - would it be clear from the emitted message what the user needs to change in the code? https://github.com/getsentry/sentry/blob/916fe38f45a23d27f2bde851e6521b66c1404299/src/sentry/utils/metrics.py#L262

**Comment by clavedeluna** [comment]
@mbyrnepr2 good catch. I'm thinking for this type of example

```
    try:
        yield current_tags
    except Exception:
        current_tags["result"] = "failure"
        raise
    else:
        current_tags["result"] = "success"
    finally:
        timing(key, time.monotonic() - start, instance, current_tags, sample_rate)
```

or perhaps any with `finally`, we should actually never raise the message. What do you think @nickdrozd ?

**Comment by nickdrozd** [comment]
I agree. The purpose of the check is to show code that can be simplified by removing an `else`. But with `finally` this cannot be done, so no warning should be raised.

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -531,7 +541,7 @@ def _is_bool_const(node: nodes.Return | nodes.Assign) -> bool:
             node.value.value, bool
         )
 
-    def _is_actual_elif(self, node: nodes.If) -> bool:
+    def _is_actual_elif(self, node: nodes.If | nodes.TryExcept) -> bool:
```
Comment (mbyrnepr2) [inline pylint/checkers/refactoring/refactoring_checker.py:544]: Is this change necessary?

**Review by mbyrnepr2 — COMMENTED** [review]
Looks solid overall. One question.

File: pylint/checkers/refactoring/refactoring_checker.py (hunk 1 above)
Comment (clavedeluna) [reply pylint/checkers/refactoring/refactoring_checker.py:544]: I believe so, because in this line https://github.com/PyCQA/pylint/pull/7888/files/341e7098da296c12fffefbfe43eb4c133426132b#diff-af1d215b95c8b3ac59218ba588186da4aefd936eef4732518196ea309c3cc788R739  a node of either type can land in there

**Review by mbyrnepr2 — APPROVED** [review]

File: pylint/checkers/refactoring/refactoring_checker.py (hunk 1 above)
Comment (mbyrnepr2) [reply pylint/checkers/refactoring/refactoring_checker.py:544]: 👍

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Can this be merged with a backport? I have lot track of whether merging with the backport label works or doesn't :sweat_smile:

**Comment by Pierre-Sassoulas** [comment]
Everything in 2.15.8 was backported automatically. The maintenance branch and the main branch were separated a long time ago so sometime we have cherry-picking issue but apart from that the backporting label is working well. (This should not be an issue anymore when we release 2.16.0). However we're only backporting false positives or crash/bug fixes so we don't add new messages in patch version so I don't think this one should be backported. (See https://pylint.pycqa.org/en/latest/user_guide/installation/upgrading_pylint.html)

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -707,23 +721,38 @@ def visit_with(self, node: nodes.With) -> None:
...
     ) -> None:
+        if isinstance(node, nodes.TryExcept) and isinstance(
+            node.parent, nodes.TryFinally
+        ):
+            # Not interested in try/except/else/finally statements.
+            return
+
         if not node.orelse:
-            # Not interested in if statements without else.
+            # Not interested in if /try statements without else.
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/refactoring_checker.py:736]:
```suggestion
            # Not interested in if/try statements without else.
```

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Review by Pierre-Sassoulas — DISMISSED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
